enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,545 @@
+            checked=False,
+        ),
+        dlf.BaseLayer(
+            dlf.TileLayer(
+                url="https://{s}.tile.opentopomap.org/{z}/{x}/{y}.png"
+            ),
+            name="Topo",
+            checked=True,
+        ),
+    ] + [
+        make_adm_overlay(
+            adm["name"],
+            adm["sql"],
+            adm["color"],
+            i+1,
+            len(GLOBAL_CONFIG["shapes_adm"])-i,
+            is_checked=adm["is_checked"]
+        )
+        for i, adm in enumerate(GLOBAL_CONFIG["shapes_adm"])
+    ] + [
+        dlf.Overlay(
+            dlf.TileLayer(
+                url=f"{TILE_PFX}/{{z}}/{{x}}/{{y}}?{qstr}",
+                opacity=1,
+            ),
+            name="Crop Suitability",
+            checked=True,
+        ),
+    ]
+
+
+@APP.callback(
+    Output("navbar-collapse", "is_open"),
+    Input("navbar-toggler", "n_clicks"),
+    State("navbar-collapse", "is_open"),
+)
+def toggle_navbar_collapse(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+def round_latLng(coord):
+    value = float(coord)
+    value = round(value, 4)
+    return value
+
+
+@APP.callback(
+    Output("hover_feature_label", "children"),
+    Input({"type": "borders_adm", "index": ALL}, "hover_feature")
+)
+def write_hover_adm_label(adm_loc):
+    location_description = "the map will return location name"
+    for i, adm in enumerate(adm_loc):
+        if adm is not None:
+            location_description = adm['geometry']['label']
+    return f'Mousing over {location_description}'
+
+
+@APP.callback(
+    Output("map_description", "children"),
+    Input("data_choice", "value"),
+)
+def write_map_description(data_choice):
+    return CONFIG["map_text"][data_choice]["description"]    
+
+@APP.callback(
+    Output("loc_marker", "position"),
+    Output("lat_input", "value"),
+    Output("lng_input", "value"),
+    Input("submit_lat_lng","n_clicks"),
+    Input("map", "click_lat_lng"),
+    State("lat_input", "value"),
+    State("lng_input", "value")
+)
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+    if dash.ctx.triggered_id == None:
+        lat = rr_mrg.precip["Y"][int(rr_mrg.precip["Y"].size/2)].values
+        lng = rr_mrg.precip["X"][int(rr_mrg.precip["X"].size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
+            lat = latitude
+            lng = longitude
+        try:
+            nearest_grid = pingrid.sel_snap(rr_mrg.precip, lat, lng)
+            lat = nearest_grid["Y"].values
+            lng = nearest_grid["X"].values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return [lat, lng], lat, lng
+
+def sel_year_season(data,target_season,target_year):
+    seasonal = data.sel(T=data['T.season']==target_season)
+    if target_season == "DJF": #need december of previous year right now this does not work
```

Comment:
Yeah, I think we need to label each day with an identifier of the season it belongs to. E.g. define the season as a start date and a length (in months, to avoid issues with leap years?), and label each day with the start date of its season, then filter on the month of that, and use the year of the start of the season as the coordinate of the resulting subset.

You don't have to invent this all from scratch. @remicousin has done a lot of it in the onset maproom.